### PR TITLE
fix erroneous leftover variable in error message

### PR DESCRIPTION
**Pull request by tdent**

### CODE REVIEW

**Comment by titodalcanton** [comment]
It should raise a `ValueError`, not log an error message. If you follow the code through, after this PR the code will happily print the error message and then crash on line 221 due to `original_gid` being undefined.

**Comment by titodalcanton** [comment]
`parser.error()` is also a reasonable choice, but I think that is more appropriate when the user gives an invalid CLI option. In this case the error depends on the contents of an input file, so a `ValueError` seems more appropriate.

**Comment by tdent** [comment]
I thought you suggested logging.error() in the previous PR. Still, I can go back to the python exception quite easily ..
So logging.error() doesn't actually cause the code to stop?

**Comment by titodalcanton** [comment]
Yeah sorry, I had suggested `logging.error()` instead of `print()`, when I should have suggested an exception right away. `logging.error()` does not stop the program.

**Comment by tdent** [comment]
fwiw, python documentation says `ValueError - Raised when an operation or function receives an argument that has the right type but an inappropriate value`

File: bin/pycbc_optimize_snr [hunk 1]
```diff
@@ -204,19 +204,19 @@ pycbc.init_logging(args.verbose)
 logging.info('Starting optimize SNR')
 
 data = {}
-for ifo in args.data_files.keys():
+ifos = list(data.keys())
+for ifo in ifos:
```
[the remark this answers is not in the document]
Comment (tdent) [reply bin/pycbc_optimize_snr:208]: urgh, not quite concentrating here

**Review by titodalcanton — APPROVED** [review]

**Pull request merged**